apps/web/src/views/Home/components/UserBanner/HarvestCard.tsx

```diff
@@ -90,8 +91,27 @@ const HarvestCard = () => {
       }
     }
 
-    onHarvestAll(v3Farms.map((farm) => farm.sendTx.tokenId))
-  }, [farmsWithStakedBalance, onHarvestAll, fetchWithCatchTxError, gasPrice, toastSuccess, t])
+    const v3Farms = (
+      farmsWithStakedBalance.filter((value) => value && 'sendTx' in value) as {
+        sendTx: {
+          to: string
+          tokenId: string
+        }
+      }[]
+    ).map((farm) => farm.sendTx.tokenId)
+    if (v3Farms.length > 0) {
+      onHarvestAll(v3Farms, onHarvestStart, onHarvestEnd)
```

Comment:
can we keep the `onHarvestStart` and `onHarvestEnd` outside the `onHarvestAll` function?

```typescript
try {
  onHarvestStart()
  await onHarvestAll(v3Farms)
} catch (e) {
  throw e
} finally {
  onHarvestEnd()
}
```